### ISSUE TITLE
Not compiling at current master `0786ddddb` state

### ISSUE THREAD

**Issue by Azgrom** [nushell/nushell#10084]
### Describe the bug

The installation script does not compiles. Here are the current commit and returned error message:
```bash
PS E:\Repos\nushell> git lg            ] 506/520: nu-command
0786ddddb (HEAD -> main, origin/main, origin/HEAD) allow `help` to return a `Type::Table` (#10082)
7617084ca Polish `CONTRIBUTING`, add Rust style (#10071)
28941f1a0 Remove global clippy `-A` from `toolkit.nu` (#10073)
43ceb3ede Remove clippy global `-A` from CI (#10072)
bffd8e4dd Pin reedline to 0.23.0 (#10070)
66023f624 Remove "let config" warning (#10068)
10fc32e3e Simplify virtualenv testing (#10035)
3148acd3a Recursively export constants from modules (#10049)
e6ce8a89b try and fix `into datetime` to accept more dt formats (#10063)
74f808129 allow `return` to return any nushell value (#10067)
2ae1de247 move 'bytes' back to commands (#10051)
028a327ce Revert "deprecate `--format` and `--list` in `into datetime` (#10017)" (#10055)
318862aad pin serde to avoid https://github.com/serde-rs/serde/issues/2538 (#10061)
9f4510f2e make the charpage optional for `std clip` (#10053)
98c7ab96b enable/update some example tests so they work again (#10058)
PS E:\Repos\nushell> .\scripts\install-all.ps1
-----------------------------------------------------------------
Installing nushell (nu) with dataframes and all the plugins
-----------------------------------------------------------------

Install nushell from local...
----------------------------------------------
  Installing nu v0.83.2 (E:\Repos\nushell)
    Updating crates.io index
   Compiling polars-arrow v0.30.0
   Compiling nu-command v0.83.2 (E:\Repos\nushell\crates\nu-command)
error[E0034]: multiple applicable items in scope
  --> C:\Users\rafae\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-arrow-0.30.0\src\array\get.rs:33:17
   |
33 |         if self.is_null_unchecked(item) {
   |                 ^^^^^^^^^^^^^^^^^ multiple `is_null_unchecked` found
   |
note: candidate #1 is defined in an impl of the trait `IsValid` for the type `A`
  --> C:\Users\rafae\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-arrow-0.30.0\src\is_valid.rs:34:5
   |
34 |     unsafe fn is_null_unchecked(&self, i: usize) -> bool {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: candidate #2 is defined in an impl of the trait `arrow2::array::Array` for the type `arrow2::array::PrimitiveArray<T>`
help: disambiguate the method for candidate #1
   |
33 |         if IsValid::is_null_unchecked(&self, item) {
   |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
help: disambiguate the method for candidate #2
   |
33 |         if arrow2::array::Array::is_null_unchecked(&self, item) {
   |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error[E0034]: multiple applicable items in scope
  --> C:\Users\rafae\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-arrow-0.30.0\src\array\get.rs:56:17
   |
56 |         if self.is_null_unchecked(item) {
   |                 ^^^^^^^^^^^^^^^^^ multiple `is_null_unchecked` found
   |
note: candidate #1 is defined in an impl of the trait `IsValid` for the type `A`
  --> C:\Users\rafae\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-arrow-0.30.0\src\is_valid.rs:34:5
   |
34 |     unsafe fn is_null_unchecked(&self, i: usize) -> bool {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: candidate #2 is defined in an impl of the trait `arrow2::array::Array` for the type `arrow2::array::BooleanArray`
help: disambiguate the method for candidate #1
   |
56 |         if IsValid::is_null_unchecked(&self, item) {
   |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
help: disambiguate the method for candidate #2
   |
56 |         if arrow2::array::Array::is_null_unchecked(&self, item) {
   |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error[E0034]: multiple applicable items in scope
  --> C:\Users\rafae\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-arrow-0.30.0\src\array\get.rs:79:17
   |
79 |         if self.is_null_unchecked(item) {
   |                 ^^^^^^^^^^^^^^^^^ multiple `is_null_unchecked` found
   |
note: candidate #1 is defined in an impl of the trait `IsValid` for the type `A`
  --> C:\Users\rafae\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-arrow-0.30.0\src\is_valid.rs:34:5
   |
34 |     unsafe fn is_null_unchecked(&self, i: usize) -> bool {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: candidate #2 is defined in an impl of the trait `arrow2::array::Array` for the type `arrow2::array::Utf8Array<O>`
help: disambiguate the method for candidate #1
   |
79 |         if IsValid::is_null_unchecked(&self, item) {
   |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
help: disambiguate the method for candidate #2
   |
79 |         if arrow2::array::Array::is_null_unchecked(&self, item) {
   |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error[E0034]: multiple applicable items in scope
   --> C:\Users\rafae\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-arrow-0.30.0\src\array\get.rs:102:17
    |
102 |         if self.is_null_unchecked(item) {
    |                 ^^^^^^^^^^^^^^^^^ multiple `is_null_unchecked` found
    |
note: candidate #1 is defined in an impl of the trait `IsValid` for the type `A`
   --> C:\Users\rafae\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-arrow-0.30.0\src\is_valid.rs:34:5
    |
34  |     unsafe fn is_null_unchecked(&self, i: usize) -> bool {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: candidate #2 is defined in an impl of the trait `arrow2::array::Array` for the type `arrow2::array::BinaryArray<O>`
help: disambiguate the method for candidate #1
    |
102 |         if IsValid::is_null_unchecked(&self, item) {
    |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
help: disambiguate the method for candidate #2
    |
102 |         if arrow2::array::Array::is_null_unchecked(&self, item) {
    |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error[E0034]: multiple applicable items in scope
   --> C:\Users\rafae\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-arrow-0.30.0\src\array\get.rs:126:17
    |
126 |         if self.is_null_unchecked(item) {
    |                 ^^^^^^^^^^^^^^^^^ multiple `is_null_unchecked` found
    |
note: candidate #1 is defined in an impl of the trait `IsValid` for the type `A`
   --> C:\Users\rafae\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-arrow-0.30.0\src\is_valid.rs:34:5
    |
34  |     unsafe fn is_null_unchecked(&self, i: usize) -> bool {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: candidate #2 is defined in an impl of the trait `arrow2::array::Array` for the type `arrow2::array::ListArray<O>`
help: disambiguate the method for candidate #1
    |
126 |         if IsValid::is_null_unchecked(&self, item) {
    |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
help: disambiguate the method for candidate #2
    |
126 |         if arrow2::array::Array::is_null_unchecked(&self, item) {
    |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error[E0034]: multiple applicable items in scope
   --> C:\Users\rafae\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-arrow-0.30.0\src\array\get.rs:150:17
    |
150 |         if self.is_null_unchecked(item) {
    |                 ^^^^^^^^^^^^^^^^^ multiple `is_null_unchecked` found
    |
note: candidate #1 is defined in an impl of the trait `IsValid` for the type `A`
   --> C:\Users\rafae\.cargo\registry\src\github.com-1ecc6299db9ec823\polars-arrow-0.30.0\src\is_valid.rs:34:5
    |
34  |     unsafe fn is_null_unchecked(&self, i: usize) -> bool {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: candidate #2 is defined in an impl of the trait `arrow2::array::Array` for the type `FixedSizeListArray`
help: disambiguate the method for candidate #1
    |
150 |         if IsValid::is_null_unchecked(&self, item) {
    |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
help: disambiguate the method for candidate #2
    |
150 |         if arrow2::array::Array::is_null_unchecked(&self, item) {
    |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

For more information about this error, try `rustc --explain E0034`.
error: could not compile `polars-arrow` due to 6 previous errors
warning: build failed, waiting for other jobs to finish...
```

### How to reproduce

Just be at commit `0786ddddb` and run the installation script

### Expected behavior

For the project to compile

### Configuration

| key                | value                                         |
| ------------------ | --------------------------------------------- |
| version            | 0.83.1                                        |
| branch             | main                                          |
| commit_hash        | 6ac3351fd13e317aacebaa8203ed01fc94cc4025      |
| build_os           | windows-x86_64                                |
| build_target       | x86_64-pc-windows-msvc                        |
| rust_version       | rustc 1.69.0 (84c898d65 2023-04-16)           |
| rust_channel       | 1.69.0-x86_64-pc-windows-msvc                 |
| cargo_version      | cargo 1.69.0 (6e9a83356 2023-04-12)           |
| build_time         | 2023-07-28 20:00:53 -03:00                    |
| build_rust_channel | release                                       |
| allocator          | standard                                      |
| features           | dataframe, default, sqlite, trash, which, zip |
| installed_plugins  |                                               |

**Comment by fdncred** [comment]
We need to update those. There was an upstream change somewhere that now requires us to use `--locked` in the cargo command line.

**Comment by amtoine** [comment]
might need to update `toolkit.nu` and `scripts/install-all.sh` as well :+1:

**Comment by fdncred** [comment]
@amtoine yup, we should update all of them. Even if the upstream gets fixed, I don't think there's any harm to "locking".

**Comment by fdncred** [comment]
I just landed #10086 which fixes this. please reopen if you still see problems after that PR.

**Comment by Azgrom** [comment]
That`s it, thank you!
